components/application-mgt/org.wso2.carbon.identity.application.mgt/src/test/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtilTest.java

```diff
@@ -79,4 +161,366 @@ public void testSpNameValidationWithCustomRegex(String appName, boolean isValidN
 
         Assert.assertEquals(ApplicationMgtUtil.isRegexValidated(appName), isValidName);
     }
+
+    @Test
+    public void testBuildPermissions() {
+
+        String[] permissions = new String[]{"permission"};
+        org.wso2.carbon.user.api.Permission[] permissionSet = ApplicationMgtUtil.buildPermissions(APPLICATION_NAME,
+                permissions);
+        assertEquals(permissionSet[0].getResourceId(), APPLICATION_NAME + "\\permission");
+    }
+
+    @DataProvider(name = "validateRolesDataProvider")
+    public Object[][] validateRolesDataProvider() {
+
+        return new Object[][]{
+                {TRUE_VALUE, TRUE},
+                {null, TRUE},
+                {FALSE_VALUE, FALSE}
+        };
+    }
+
+    @Test(dataProvider = "validateRolesDataProvider")
+    public void testValidateRoles(String allowRoleValidationProperty, Boolean expected) {
+
+        mockStatic(IdentityUtil.class);
+        when(IdentityUtil.getProperty(ENABLE_APPLICATION_ROLE_VALIDATION_PROPERTY)).
+                thenReturn(allowRoleValidationProperty);
+
+        assertEquals(ApplicationMgtUtil.validateRoles(), expected);
+    }
+
+    @DataProvider(name = "userAuthorizeDataProvider")
+    public Object[][] userAuthorizeDataProvider() {
+
+        String[] userRoles1 = {ROLE_NAME, ""};
+        String[] userRoles2 = {APPLICATION_NAME, ""};
+
+        return new Object[][]{
+                {APPLICATION_NAME, USERNAME, TRUE_VALUE, userRoles1, 1, TRUE },
+                {APPLICATION_NAME, USERNAME, "", userRoles2, 2, FALSE },
+                {APPLICATION_NAME, USERNAME, FALSE_VALUE, userRoles1, 1, TRUE }
```

Comment:
Is it possible to add some more cases with null values as well?

Reply:
Added.